doc/source/user_guide/text.rst

```diff
@@ -262,25 +262,30 @@ i.e., from the end of the string to the beginning of the string:
                    '', np.nan, 'CABA', 'dog', 'cat'],
                   dtype="string")
    s3
-   s3.str.replace('^.a|dog', 'XX-XX ', case=False)
+   s3.str.replace('^.a|dog', 'XX-XX ', case=False, regex=True)
 
-Some caution must be taken to keep regular expressions in mind! For example, the
-following code will cause trouble because of the regular expression meaning of
-``$``:
+Some caution must be taken when dealing with regular expressions! The current behavior
+is to treat single character patterns as literal strings, even when ``regex`` is set
+to ``True``. (This behavior is deprecated and will be removed in a future version so
+that the ``regex`` keyword is always respected.) For example, the following code will
+cause trouble because of the regular expression meaning of ``$``:
 
 .. ipython:: python
 
    # Consider the following badly formatted financial data
    dollars = pd.Series(['12', '-$10', '$10,000'], dtype="string")
 
-   # This does what you'd naively expect:
-   dollars.str.replace('$', '')
+   # Here $ is treated as a literal character
+   dollars.str.replace('$', '', regex=True)
```

Comment:
but will it also still be interpreted as a literal character when `regex=True` is specified? Or only for the default of passing nothing  (which now becomes regex=None)

Reply:
Currently it will since single character regexes are ignored. I think once the default is switched then this should also be fixed at the same time (regex=True always meaning interpret as regex even for single characters).